File: lib/features/customer_service/controllers/customer_service_controller.dart
import 'package:flutter/material.dart';
import 'package:get/get.dart';
import '../models/cs_model.dart';
import '../services/cs_provider.dart';

class CustomerServiceController extends GetxController {
  final CsProvider _provider = Get.put(CsProvider());

  var currentViewIndex = 0.obs;
  var isLoading = false.obs;
  var isAdminMode = false.obs;

  var faqList = <FaqModel>[].obs;
  var myInquiries = <TicketModel>[].obs;
  var adminInquiries = <TicketModel>[].obs;

  var selectedInquiry = Rxn<TicketModel>();
  var selectedFaq = Rxn<FaqModel>();

  final titleController = TextEditingController();
  final contentController = TextEditingController();
  final answerController = TextEditingController();

  @override
  void onInit() {
    super.onInit();
    fetchData();
  }

  void fetchData() async {
    isLoading.value = true;
    try {
      await Future.wait([
        fetchFaqs(),
        fetchMyTickets(),
        if (isAdminMode.value) fetchAdminTickets(),
      ]);
    } finally {
      isLoading.value = false;
    }
  }

  Future<void> fetchFaqs() async {
    final response = await _provider.getFaqs();
    if (!response.status.hasError && response.body != null) {
      List<dynamic> data = response.body;
      faqList.value = data.map((json) => FaqModel.fromJson(json)).toList();
    }
  }

  Future<void> fetchMyTickets() async {
    final response = await _provider.getMyTickets();
    if (!response.status.hasError && response.body != null) {
      List<dynamic> data = response.body;
      var list = data.map((json) => TicketModel.fromJson(json)).toList();
      list.sort((a, b) => b.id.compareTo(a.id));
      myInquiries.value = list;
    }
  }

  Future<void> fetchAdminTickets() async {
    final response = await _provider.getAdminTickets();
    if (!response.status.hasError && response.body != null) {
      List<dynamic> data = response.body;
      adminInquiries.value = data.map((json) => TicketModel.fromJson(json)).toList();
    }
  }

  void toggleAdminMode() {
    isAdminMode.value = !isAdminMode.value;
    fetchData();
    currentViewIndex.value = 0;
  }

  void changeView(int index) {
    if (index == 1) isAdminMode.value ? fetchAdminTickets() : fetchMyTickets();
    currentViewIndex.value = index;
  }

  void viewDetail(TicketModel inquiry) {
    selectedInquiry.value = inquiry;
    changeView(3);
  }

  void viewFaqDetail(FaqModel faq) {
    selectedFaq.value = faq;
    changeView(4);
  }

  Future<void> submitInquiry() async {
    if (titleController.text.isEmpty || contentController.text.isEmpty) {
      Get.snackbar('알림', '제목과 내용을 입력해주세요.');
      return;
    }
    isLoading.value = true;
    final response = await _provider.createTicket(titleController.text, contentController.text);
    isLoading.value = false;
    if (!response.status.hasError) {
      Get.snackbar('성공', '문의가 등록되었습니다.');
      titleController.clear();
      contentController.clear();
      await fetchMyTickets();
      changeView(1);
    }
  }

  Future<void> submitAdminAnswer() async {
    if (answerController.text.isEmpty || selectedInquiry.value == null) return;
    isLoading.value = true;
    final response = await _provider.postAdminAnswer(selectedInquiry.value!.id, answerController.text);
    isLoading.value = false;
    if (!response.status.hasError) {
      Get.snackbar('성공', '답변이 등록되었습니다.');
      answerController.clear();
      fetchAdminTickets();
      changeView(1);
    }
  }

  @override
  void onClose() {
    titleController.dispose();
    contentController.dispose();
    answerController.dispose();
    super.onClose();
  }
}

File: lib/features/CustomerServiceController.java
import 'package:get/get.dart';

class CustomerServiceController extends GetxController {
  var faqs = <Map<String, String>>[].obs;
  var inquiries = <Map<String, String>>[].obs;

  @override
  void onInit() {
    super.onInit();
    faqs.value = [
      {"question": "책 등록 방법", "answer": "홈페이지에서 '책 등록' 버튼 클릭 후 입력"},
      {"question": "책 검색 방법", "answer": "검색창에 제목 입력 후 검색"},
      {"question": "리뷰 작성 방법", "answer": "도서 상세 페이지에서 리뷰 작성"},
      {"question": "평점 남기기", "answer": "책 상세 페이지에서 별점 입력"},
      {"question": "북마크 추가 방법", "answer": "책 상세 페이지에서 북마크 버튼 클릭"},
      {"question": "위시리스트 추가", "answer": "도서 상세 페이지에서 위시 버튼 클릭"},
      {"question": "문의 답변 확인", "answer": "내 문의 내역에서 확인 가능"},
    ];

    inquiries.value = [
      {"title": "책 등록 문의", "status": "waiting"},
      {"title": "리뷰 등록 문의", "status": "answered"},
    ];
  }

  void addInquiry(String title) {
    inquiries.add({"title": title, "status": "waiting"});
  }
}
